### ISSUE TITLE
qute-props with 'consistent-as-needed' doesn't behave as expected

### ISSUE THREAD

**Issue by medikoo** [eslint/eslint#12619]
It doesn't recognize quoted method shorthands, which is unexpected

**Tell us about your environment**

* **ESLint Version:** v6.7.1
* **Node Version:** v13.2.0
* **npm Version:** v6.3.1

**What parser (default, Babel-ESLint, etc.) are you using?**

None

**Please show your full configuration:**

https://github.com/medikoo/eslint-config-medikoo/blob/219731aa5d9d9596dccb2341f070be6224c4d04a/index.js#102

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js

const obj = { "non standard method name"() {}, "propertyName": {} };
```

**What did you expect to happen?**

No ESlint error reported

**What actually happened? Please include the actual, raw output from ESLint.**
```
error  Properties shouldn't be quoted as all quotes are redundant  quote-props
```

**Are you willing to submit a pull request to fix this bug?**

No

**Comment by mdjermanovic** [comment]
Hi @medikoo, thanks for the bug report!

(configuration is: `"quote-props": ["error", "consistent-as-needed"]`)

This rule completely ignores methods (as well as shorthand properties and computed properties).

Consistency is checked and enforced between the remaining properties only. In the example, `"propertyName"` is the only one.

Since the (necessarily quoted) method in the example doesn't count at all, `consistent-as-needed` option reports the same error as it would be reported on:

```js
const obj = { "propertyName": {} };
```

By looking at the rule history, the first change was to ignore methods for the `"always"` option only (PR #2286, issue #2251) because quotes are not common in method names. At the time, the rule didn't have 'consistent' options.

After #3558, methods are ignored for all options; the rule doesn't enforce anything on methods, and quoted or unquoted methods do not affect `"consistent"` and `"consistent-as-needed"` logic for other 'normal' (colon) properties in any way. Though, the PR was mainly about computed and shorthand properties, I couldn't find a discussion about methods.

Interestingly, getters and setters, which have similar syntax as methods, are treated as 'normal' properties.

Anyway, I think that any change in this behavior would be a breaking change now. If this rule should be enforced on methods as well, or at least include methods in the 'consistent' logic, the best solution would be, in my opinion, to make a proposal for a new option or options.

**Comment by medikoo** [comment]
@mdjermanovic while understand, why it doesn't work like that now, I can hardly believe anyone would expect the output as following with `'consistent-as-needed'`

```javascript
const obj = { "non standard method name"() {}, propertyName: {} };
```

And this is the way it works now.

Maybe it'll be good to create some poll, to confirm on that.

Assuming that 99% of users expects the following:

```
const obj = { "non standard method name"() {}, "propertyName": {} };
```

The right move is to just change the current behavior. It can be marked as breaking, although if output is seen as intended I don't think this should be considered breaking

**Comment by mdjermanovic** [comment]
I agree that this particular example might be unexpected, but what should be expected for the following code when the option is `'consistent-as-needed'`:

```js
const obj = { 
    methodName() {},
    "non standard method name"() {},
    propertyName: {}
};
```

If the rule should consider methods in its logic, then it should also require/disallow quotes in method names?

```js
const obj = { 
    "methodName"() {}, // required for consistency
    "non standard method name"() {},
    "propertyName": {} 
};

const obj2 = { 
    "methodName"() {}, // required for consistency
    "non standard property name": {} 
};
```

Then, it should also require quotes in all method names when the option is `'always'` as it probably doesn't make sense to sometimes target methods and sometimes not.

This looks to me like a breaking change.

**Comment by medikoo** [comment]
Other thing I forgot to mention, is that the way it currently works in ESLint, is not on par with how same rule works in Prettier: https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEEBGArTAXk2EwB0QppMMBDKAE1oCcHMBbOGACwjalqcKACgCUJAL4AaTAAdmEWXGYwAngDlBcJJMwSA3CCkhFMAJbpkoFgoDuABRYI0yELQBuEMwyMgczWjAAay4AZVlAsygAc2QYZgBXOGNuGHYAGwB1bjN4NAiwOFDnXLN3XNVXMDQXYyi0ZRh7AOj2WmQAM1p0huM8NAAPACEA4LCtABkouE7u3pB+gdCo6PS4AEUEiHhZnuSQCOYG5lccWhw4dN95KJhM7x5kAA4ABmN5CAbMgNlXeThju4ZsYAI5beDNRQuFCQKBoMwYBAwXzMOBgsyo5q0VrtJBdPbGBrsMxxRL7eExNabbYzPFzfYwc73BiPJAAJmM8VoZnSKwAwhB2G1XFQoMCQAkGgAVc7Q-HzdxJACSjCRoTAzDMshgAEFGKE1Gtdg0JBIgA

Still Prettier users will most likely turn off that rule in ESLint

**Comment by mdjermanovic** [comment]
> Other thing I forgot to mention, is that the way it currently works in ESLint, is not on par with how same rule works in Prettier

Prettier's `consistent` seems to work as described in [the comment](https://github.com/eslint/eslint/issues/12619#issuecomment-560023149) above, i.e. treat methods in same way as other properties.

This also means that it requires quotes in all method names whenever at least one method or  property has necessary quotes, so the following code is invalid:

```js
const obj2 = { 
    methodName() {}, // error, should be fixed to "methodName"() {}
    "non standard property name": {} 
};
```

This behavior as a user's preference seems reasonable to me, but it has to be behind an additional option because it significantly changes the current behavior and might be undesirable for other users. I still think it would be better to open a rule enhancement proposal, e.g., for an additional boolean option to include methods.
